Tests/SwiftSyntaxMacroExpansionTest/AttributeRemoverTests.swift

```diff
@@ -10,505 +10,484 @@
 //
 //===----------------------------------------------------------------------===//
 
-//==========================================================================//
-// IMPORTANT: The macros defined in this file are intended to test the      //
-// behavior of MacroSystem. Many of them do not serve as good examples of   //
-// how macros should be written. In particular, they often lack error       //
-// handling because it is not needed in the few test cases in which these   //
-// macros are invoked.                                                      //
-//==========================================================================//
-
-import SwiftDiagnostics
+import SwiftParser
 import SwiftSyntax
-import SwiftSyntaxMacroExpansion
-import SwiftSyntaxMacros
-import SwiftSyntaxMacrosTestSupport
 import XCTest
-
-fileprivate struct NoOpPeerMacro: PeerMacro {
-  static func expansion(
-    of node: AttributeSyntax,
-    providingPeersOf declaration: some DeclSyntaxProtocol,
-    in context: some MacroExpansionContext
-  ) throws -> [DeclSyntax] {
-    return []
+import _SwiftSyntaxTestSupport
+@_spi(AttributeRemover) import SwiftSyntaxMacroExpansion
+
+fileprivate func assertSyntaxRemovingTestAttributes(
+  of originalSource: () throws -> String,
+  reduction expectedReducedSource: () -> String,
+  file: StaticString = #filePath,
+  line: UInt = #line
+) {
+  let attributesToRemove = ["@Test"].map(AttributeSyntax.init(stringLiteral:))
+
+  do {
+    //  let reducedSource = WrappedAttributeRemover(attributesToRemove: attributesToRemove)
+    let reducedSource = AttributeRemover(
+      attributesToRemove: attributesToRemove,
+      attributeEquals: { $0.trimmedDescription == $1.trimmedDescription }
+    )
+      .rewrite(
+        Parser.parse(
+          source: try originalSource()
+        )
+      )
+
+    assertStringsEqualWithDiff(
+      reducedSource.description.trimmingCharacters(in: .newlines),
+      expectedReducedSource().trimmingCharacters(in: .newlines),
+      "Attribute removal did not produce the expected reduced source",
+      additionalInfo: """
+      Actual reduced source:
+      \(reducedSource)
+      """,
+      file: file,
+      line: line
+    )
+  } catch {
+    XCTFail("Threw error: \(error)", file: file, line: line)
```

Comment:
If you made the inputs normal parameters instead of closures, I think there’s no need for the `do {} catch {}` anymore.